### PR TITLE
py-yt: 3.6.1

### PR DESCRIPTION
**Pull request by ax3l**
The latest bugfix release of yt-project fixes issues with matplotlib version 3.3.0+.

Refs.:
- https://github.com/yt-project/yt/issues/2752
- https://github.com/yt-project/yt/pull/2754
- https://github.com/yt-project/yt/pull/2961

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 1]
```diff
@@ -46,6 +48,7 @@ class PyYt(PythonPackage):
     depends_on("py-ipython", type=('build', 'run'))
     depends_on("py-ipython@:6.99", type=('build', 'run'), when="^python@:2.99")
     depends_on("py-matplotlib", type=('build', 'run'))
+    depends_on("py-matplotlib@:3.2.2", type=('build', 'run'), when="@:3.6.0")
     depends_on("py-numpy", type=('build', 'run'))
     depends_on("py-scipy", type=('build', 'run'), when="+scipy")
     depends_on("py-setuptools", type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: It looks like Python 2.7:2.8,3.5: is now required. You should also add type=build/run. Also, cython 0.24+ is needed. Many other dependencies are also missing versions:
```python
    install_requires=[                                                                   
        'matplotlib>=1.5.3',                                                             
        'setuptools>=19.6',                                                              
        'sympy>=1.0',                                                                    
        'numpy>=1.10.4',                                                                 
        'IPython>=1.0',                                                                  
    ],        
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-yt/package.py (hunk 1 above)
Comment (ax3l) [reply var/spack/repos/builtin/packages/py-yt/package.py]: Thank you, fixed.

**Comment by ax3l** [comment]
@qobilidop I saw your PR in #11350 - do you want to be listed as package maintainer for `yt` on Spack?

**Review by adamjstewart — DISMISSED** [review]
I don't see where the asotrpy/h5py/scipy/rockstar variants come from or where there deps are specified, but everything else looks fine.

**Comment by ax3l** [comment]
I think someone specified them randomly from the test requirements: https://github.com/yt-project/yt/tree/yt-3.6.1/tests

But yt is too large and my contributions & understanding too small to commit to doing larger changes on this package than fixing the bug in front of me :D

**Comment by qobilidop** [comment]
> @qobilidop I saw your PR in #11350 - do you want to be listed as package maintainer for `yt` on Spack?

Sure, that's fine. And thanks!

**Comment by ax3l** [comment]
Thank you as well :+1: :tada:

**Review by alalazo — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

**Comment by ax3l** [comment]
I think the CI error is probably unrelated:
```
                if os.path.samestat(orig_st, os.fstat(fd)):
                    _rmtree_safe_fd(fd, path, onerror)
                    try:
>                       os.rmdir(path)
E                       OSError: [Errno 66] Directory not empty: '/Users/runner/work/spack/spack/lib/spack/spack/__pycache__'

../../../hostedtoolcache/Python/3.8.8/x64/lib/python3.8/shutil.py:720: OSError
```
?

**Pull request merged**